Cover text_selection tests with leak tracking. [prod-leak-fix]

ksokolovskyi opened the pull request:
### Description
 - This PR changes `testWidgets` by `testWidgetsWithLeakTracking` in number of `flutter/test/widgets` tests according to the https://github.com/dart-lang/leak_tracker/issues/134.
- Fixes leak in `RenderEditable`.

### Tests
- Updates `widgets/text_selection_test.dart` to use `testWidgetsWithLeakTracking`;
- Updates `widgets/text_selection_toolbar_layout_delegate_test.dart ` to use `testWidgetsWithLeakTracking`.

## Pre-launch Checklist

- [X] I read the [Contributor Guide] and followed the process outlined there for submitting PRs.
- [X] I read the [Tree Hygiene] wiki page, which explains my responsibilities.
- [X] I read and followed the [Flutter Style Guide], including [Features we expect every widget to implement].
- [X] I signed the [CLA].
- [X] I listed at least one issue that this PR fixes in the description above.
- [ ] I updated/added relevant documentation (doc comments with `///`).
- [X] I added new tests to check the change I am making, or this PR is [test-exempt].
- [X] All existing and new tests are passing.

ksokolovskyi committed:
Update text_selection_test.dart with leak tracking and fix leak in RenderEditable

ksokolovskyi commented:
cc @polina-c

ksokolovskyi committed:
Merge branch 'master' into text-selection-leak-tracking

polina-c commented on packages/flutter/lib/src/rendering/editable.dart:
```diff
@@ -405,6 +407,7 @@ class RenderEditable extends RenderBox with RelayoutWhenSystemFontsChangeMixin,
     _selectionPainter.dispose();
     _caretPainter.dispose();
     _textPainter.dispose();
+    _internalShowCursor?.dispose();
```
There is `set showCursor(ValueNotifier<bool> value) {`. It makes things little more complicated. Should the setter dispose previous value?
Instead of creating new member `_internalShowCursor`, should we create flag `_disposeShowCursor` and update it every time _showCursor is assigned?

ksokolovskyi replied:
@polina-c, thanks for the review. 
I see what you mean, but we need to care only about the first internal `ValueNotifier`, which gets created on `RenderEditable` creation if  `showCursor` is not passed. All next `showCursor` `ValueNotifier`s will be set using a setter, and their disposal is not our responsibility. 
Am I missing something?)

polina-c replied:
True, however, if setter is invoked, should we dispose the previous value right away and null out all references to it, without waiting for disposal of the entire class?

ksokolovskyi replied:
@polina-c great thought! 
Will do it that way, thanks :)

ksokolovskyi committed:
Fix

ksokolovskyi committed:
Merge branch 'master' into text-selection-leak-tracking

ksokolovskyi committed:
Merge branch 'master' into text-selection-leak-tracking

ksokolovskyi committed:
Removed leakTrackingTestConfig

ksokolovskyi committed:
Merge branch 'master' into text-selection-leak-tracking

ksokolovskyi committed:
Merge branch 'master' into text-selection-leak-tracking

polina-c committed:
Merge branch 'master' into text-selection-leak-tracking

liamappelbe committed:
Reland `coverableLineCache` optimisation (#136851) (#137385)

Relands #136851, which was rolled back in #137121

package:coverage has been rolled, so the breakages should be fixed.
Also, in this reland I've changed the `coverableLineCache` parameter to
be optional, which is safer.

christopherfujino committed:
[flutter_tool] change print when selecting developer identity for code signing (#136978)

Fixes https://github.com/flutter/flutter/issues/136984

christopherfujino committed:
[flutter_tools] ensure processUtils reports exit code in ProcessExceptions (#136672)

Help to debug situations like: https://github.com/flutter/flutter/issues/135982

p-mazhnik committed:
[web] cache the base URL as root index.html (#136594)

Fixes https://github.com/flutter/flutter/issues/136593

Caching of the base url was introduced in https://github.com/flutter/flutter/pull/53666 but resources can contain multiple `index.html` files, and currently hash of the **latest** asset will be assigned to the base url, which is not necessarily the root index.html

engine-flutter-autoroll committed:
Manual roll Flutter Engine from 84dcb4fb9301 to 4f87c20b3c34 (24 revisions) (#137570)

Manual roll requested by zra@google.com

https://github.com/flutter/engine/compare/84dcb4fb9301...4f87c20b3c34

2023-10-30 skia-flutter-autoroll@skia.org Roll Dart SDK from
5a666e8d8259 to 85e149f894df (6 revisions) (flutter/engine#47470)
2023-10-30 dacoharkes@google.com [deps] Stop moving NDK folder inside
Android SDK (flutter/engine#47454)
2023-10-30 jonahwilliams@google.com Revert "[Impeller] remove image
upload workarounds." (flutter/engine#47402)
2023-10-30 skia-flutter-autoroll@skia.org Roll Skia from 89250a0bd6bb to
817fca468240 (3 revisions) (flutter/engine#47465)
2023-10-30 skia-flutter-autoroll@skia.org Roll Skia from 006424658682 to
89250a0bd6bb (3 revisions) (flutter/engine#47463)
2023-10-30 chinmaygarde@google.com Migrate inja, sqlite, libtess2 to
//flutter/third_party. (flutter/engine#47408)
2023-10-30 zanderso@users.noreply.github.com Revert "Roll Dart SDK from
5a666e8d8259 to 38e8459ca2aa (4 revisions)" (flutter/engine#47459)
2023-10-30 skia-flutter-autoroll@skia.org Roll Skia from 09818d8c9d3c to
006424658682 (1 revision) (flutter/engine#47458)
2023-10-30 skia-flutter-autoroll@skia.org Roll Skia from 6e9cd1b79105 to
09818d8c9d3c (1 revision) (flutter/engine#47456)
2023-10-30 kjlubick@users.noreply.github.com Remove call to
SkTypeface::MakeDefault (flutter/engine#47344)
2023-10-30 skia-flutter-autoroll@skia.org Roll Fuchsia Linux SDK from
LbIVijDVwcB56_wgv... to zf2xLsbeoVrbnEDiH... (flutter/engine#47452)
2023-10-30 skia-flutter-autoroll@skia.org Roll Skia from c732ec7f5d74 to
6e9cd1b79105 (2 revisions) (flutter/engine#47451)
2023-10-30 skia-flutter-autoroll@skia.org Roll Skia from e45946f89a8b to
c732ec7f5d74 (2 revisions) (flutter/engine#47448)
2023-10-30 skia-flutter-autoroll@skia.org Roll Fuchsia Mac SDK from
OJfJYAvhpAkE-kvbY... to jCjz-imbNBDjiRVa1... (flutter/engine#47447)
2023-10-30 skia-flutter-autoroll@skia.org Roll Skia from 7545d1468268 to
e45946f89a8b (1 revision) (flutter/engine#47446)
2023-10-29 skia-flutter-autoroll@skia.org Roll Skia from e224e1dc5e2c to
7545d1468268 (1 revision) (flutter/engine#47444)
2023-10-29 skia-flutter-autoroll@skia.org Roll Fuchsia Linux SDK from
v-9_Di17lm_C_mrpZ... to LbIVijDVwcB56_wgv... (flutter/engine#47443)
2023-10-29 skia-flutter-autoroll@skia.org Roll Fuchsia Mac SDK from
s0pbwY5MGxyq0tNrT... to OJfJYAvhpAkE-kvbY... (flutter/engine#47439)
2023-10-29 skia-flutter-autoroll@skia.org Roll Skia from d29cc3fe182f to
e224e1dc5e2c (1 revision) (flutter/engine#47438)
2023-10-29 skia-flutter-autoroll@skia.org Roll Fuchsia Linux SDK from
nreLekedxzmcdb3ww... to v-9_Di17lm_C_mrpZ... (flutter/engine#47436)
2023-10-29 skia-flutter-autoroll@skia.org Roll Fuchsia Mac SDK from
qiHQ7aarDQb_1Rq3G... to s0pbwY5MGxyq0tNrT... (flutter/engine#47431)
2023-10-28 skia-flutter-autoroll@skia.org Roll Fuchsia Linux SDK from
ESkamdoXWIwkdWdP-... to nreLekedxzmcdb3ww... (flutter/engine#47429)
2023-10-28 skia-flutter-autoroll@skia.org Roll Fuchsia Mac SDK from
9YSl2rN0sj5YpYNdQ... to qiHQ7aarDQb_1Rq3G... (flutter/engine#47428)
2023-10-28 skia-flutter-autoroll@skia.org Roll Dart SDK from
5a666e8d8259 to 38e8459ca2aa (4 revisions) (flutter/engine#47426)

Also rolling transitive DEPS:
  fuchsia/sdk/core/linux-amd64 from ESkamdoXWIwk to zf2xLsbeoVrb
  fuchsia/sdk/core/mac-amd64 from 9YSl2rN0sj5Y to jCjz-imbNBDj

If this roll has caused a breakage, revert this CL and stop the roller
using the controls here:
https://autoroll.skia.org/r/flutter-engine-flutter-autoroll
Please CC aaclarke@google.com,rmistry@google.com,zra@google.com on the
revert to ensure that a human
is aware of the problem.

To file a bug in Flutter:
https://github.com/flutter/flutter/issues/new/choose

To report a problem with the AutoRoller itself, please file a bug:
https://issues.skia.org/issues/new?component=1389291&template=1850622

Documentation for the AutoRoller is here:
https://skia.googlesource.com/buildbot/+doc/main/autoroll/README.md

engine-flutter-autoroll committed:
Roll Flutter Engine from 4f87c20b3c34 to 64b0e07e2259 (4 revisions) (#137582)

https://github.com/flutter/engine/compare/4f87c20b3c34...64b0e07e2259

2023-10-30 skia-flutter-autoroll@skia.org Roll Fuchsia Mac SDK from jCjz-imbNBDjiRVa1... to sBjtpAHoetjF77qxg... (flutter/engine#47482)
2023-10-30 skia-flutter-autoroll@skia.org Roll Dart SDK from 85e149f894df to c8f75afabaf1 (1 revision) (flutter/engine#47479)
2023-10-30 zanderso@users.noreply.github.com Update package:equatable pin (flutter/engine#47475)
2023-10-30 matanlurey@users.noreply.github.com [Impeller] Switch from `glBlitFramebuffer` to implicit MSAA resolution. (flutter/engine#47282)

Also rolling transitive DEPS:
  fuchsia/sdk/core/mac-amd64 from jCjz-imbNBDj to sBjtpAHoetjF

If this roll has caused a breakage, revert this CL and stop the roller
using the controls here:
https://autoroll.skia.org/r/flutter-engine-flutter-autoroll
Please CC aaclarke@google.com,rmistry@google.com,zra@google.com on the revert to ensure that a human
is aware of the problem.

To file a bug in Flutter: https://github.com/flutter/flutter/issues/new/choose

To report a problem with the AutoRoller itself, please file a bug:
https://issues.skia.org/issues/new?component=1389291&template=1850622

Documentation for the AutoRoller is here:
https://skia.googlesource.com/buildbot/+doc/main/autoroll/README.md

engine-flutter-autoroll committed:
Roll Flutter Engine from 64b0e07e2259 to 44309c9f385c (1 revision) (#137584)

https://github.com/flutter/engine/compare/64b0e07e2259...44309c9f385c

2023-10-30 skia-flutter-autoroll@skia.org Roll Fuchsia Linux SDK from zf2xLsbeoVrbnEDiH... to 0D7msIyC3p_BdEY03... (flutter/engine#47490)

Also rolling transitive DEPS:
  fuchsia/sdk/core/linux-amd64 from zf2xLsbeoVrb to 0D7msIyC3p_B

If this roll has caused a breakage, revert this CL and stop the roller
using the controls here:
https://autoroll.skia.org/r/flutter-engine-flutter-autoroll
Please CC aaclarke@google.com,rmistry@google.com,zra@google.com on the revert to ensure that a human
is aware of the problem.

To file a bug in Flutter: https://github.com/flutter/flutter/issues/new/choose

To report a problem with the AutoRoller itself, please file a bug:
https://issues.skia.org/issues/new?component=1389291&template=1850622

Documentation for the AutoRoller is here:
https://skia.googlesource.com/buildbot/+doc/main/autoroll/README.md

engine-flutter-autoroll committed:
Roll Flutter Engine from 44309c9f385c to 17eff7ab6299 (2 revisions) (#137587)

https://github.com/flutter/engine/compare/44309c9f385c...17eff7ab6299

2023-10-31 skia-flutter-autoroll@skia.org Roll Dart SDK from c8f75afabaf1 to 9cc76cb23b8c (1 revision) (flutter/engine#47498)
2023-10-30 chris@bracken.jp [macOS] Wrap external texture tests in autorelease pool (flutter/engine#47492)

If this roll has caused a breakage, revert this CL and stop the roller
using the controls here:
https://autoroll.skia.org/r/flutter-engine-flutter-autoroll
Please CC aaclarke@google.com,rmistry@google.com,zra@google.com on the revert to ensure that a human
is aware of the problem.

To file a bug in Flutter: https://github.com/flutter/flutter/issues/new/choose

To report a problem with the AutoRoller itself, please file a bug:
https://issues.skia.org/issues/new?component=1389291&template=1850622

Documentation for the AutoRoller is here:
https://skia.googlesource.com/buildbot/+doc/main/autoroll/README.md

engine-flutter-autoroll committed:
Roll Flutter Engine from 17eff7ab6299 to b4de734eb78e (1 revision) (#137594)

https://github.com/flutter/engine/compare/17eff7ab6299...b4de734eb78e

2023-10-31 skia-flutter-autoroll@skia.org Roll Dart SDK from 9cc76cb23b8c to 2a80a8adf898 (1 revision) (flutter/engine#47501)

If this roll has caused a breakage, revert this CL and stop the roller
using the controls here:
https://autoroll.skia.org/r/flutter-engine-flutter-autoroll
Please CC aaclarke@google.com,rmistry@google.com,zra@google.com on the revert to ensure that a human
is aware of the problem.

To file a bug in Flutter: https://github.com/flutter/flutter/issues/new/choose

To report a problem with the AutoRoller itself, please file a bug:
https://issues.skia.org/issues/new?component=1389291&template=1850622

Documentation for the AutoRoller is here:
https://skia.googlesource.com/buildbot/+doc/main/autoroll/README.md

engine-flutter-autoroll committed:
Roll Flutter Engine from b4de734eb78e to 5ca5cbb6efdb (1 revision) (#137610)

https://github.com/flutter/engine/compare/b4de734eb78e...5ca5cbb6efdb

2023-10-31 skia-flutter-autoroll@skia.org Roll Fuchsia Mac SDK from sBjtpAHoetjF77qxg... to rg_Ici1tgAVF93cn9... (flutter/engine#47502)

Also rolling transitive DEPS:
  fuchsia/sdk/core/mac-amd64 from sBjtpAHoetjF to rg_Ici1tgAVF

If this roll has caused a breakage, revert this CL and stop the roller
using the controls here:
https://autoroll.skia.org/r/flutter-engine-flutter-autoroll
Please CC rmistry@google.com,zra@google.com on the revert to ensure that a human
is aware of the problem.

To file a bug in Flutter: https://github.com/flutter/flutter/issues/new/choose

To report a problem with the AutoRoller itself, please file a bug:
https://issues.skia.org/issues/new?component=1389291&template=1850622

Documentation for the AutoRoller is here:
https://skia.googlesource.com/buildbot/+doc/main/autoroll/README.md

engine-flutter-autoroll committed:
Roll Flutter Engine from 5ca5cbb6efdb to 11a569d97a26 (1 revision) (#137612)

https://github.com/flutter/engine/compare/5ca5cbb6efdb...11a569d97a26

2023-10-31 skia-flutter-autoroll@skia.org Roll Fuchsia Linux SDK from 0D7msIyC3p_BdEY03... to LCfhx_lTRJI51G0zc... (flutter/engine#47503)

Also rolling transitive DEPS:
  fuchsia/sdk/core/linux-amd64 from 0D7msIyC3p_B to LCfhx_lTRJI5

If this roll has caused a breakage, revert this CL and stop the roller
using the controls here:
https://autoroll.skia.org/r/flutter-engine-flutter-autoroll
Please CC rmistry@google.com,zra@google.com on the revert to ensure that a human
is aware of the problem.

To file a bug in Flutter: https://github.com/flutter/flutter/issues/new/choose

To report a problem with the AutoRoller itself, please file a bug:
https://issues.skia.org/issues/new?component=1389291&template=1850622

Documentation for the AutoRoller is here:
https://skia.googlesource.com/buildbot/+doc/main/autoroll/README.md

dcharkes committed:
Fix formatting (#137613)

Badly formatted code causes distraction when reading, and costs people energy when understanding code.

engine-flutter-autoroll committed:
Roll Flutter Engine from 11a569d97a26 to 6b9b806151ed (1 revision) (#137617)

https://github.com/flutter/engine/compare/11a569d97a26...6b9b806151ed

2023-10-31 skia-flutter-autoroll@skia.org Roll Dart SDK from 2a80a8adf898 to 776fad25647a (1 revision) (flutter/engine#47505)

If this roll has caused a breakage, revert this CL and stop the roller
using the controls here:
https://autoroll.skia.org/r/flutter-engine-flutter-autoroll
Please CC rmistry@google.com,zra@google.com on the revert to ensure that a human
is aware of the problem.

To file a bug in Flutter: https://github.com/flutter/flutter/issues/new/choose

To report a problem with the AutoRoller itself, please file a bug:
https://issues.skia.org/issues/new?component=1389291&template=1850622

Documentation for the AutoRoller is here:
https://skia.googlesource.com/buildbot/+doc/main/autoroll/README.md

whesse committed:
Fix divide-by-zero crash in animation_bench benchmark (#137539)

If the benchmark runs out of time before it closes the drawer it is animating, it tries to divide by zero when computing the time per frame.

Don't report time per frame for activities with zero frames. This likely only happens for the close frame action, but guards are added to all time per frame computations in this benchmark.

engine-flutter-autoroll committed:
Roll Packages from 2af69541a214 to c9fec61b6cde (12 revisions) (#137623)

https://github.com/flutter/packages/compare/2af69541a214...c9fec61b6cde

2023-10-31 stuartmorgan@google.com [image_picker] Prevent multiple active calls on iOS (flutter/packages#5272)
2023-10-30 34871572+gmackall@users.noreply.github.com [in_app_purchase_android] Add missing response code to BillingResponse enum (flutter/packages#5120)
2023-10-30 49699333+dependabot[bot]@users.noreply.github.com Bump github/codeql-action from 2.22.4 to 2.22.5 (flutter/packages#5263)
2023-10-30 john@johnmccutchan.com Bump play-service-maps dependency from version 18.1.0 to 18.2.0 (flutter/packages#5243)
2023-10-30 engine-flutter-autoroll@skia.org Roll Flutter from a4ec6278b1a4 to e12d1a798c49 (3 revisions) (flutter/packages#5267)
2023-10-30 43054281+camsim99@users.noreply.github.com [camerax] Loosen restrictions of fallback strategies for choosing resolutions (flutter/packages#5239)
2023-10-30 stuartmorgan@google.com [pigeon] Don't wrap non-nullable primitives in Obj-C (flutter/packages#5214)
2023-10-30 1838962+tatsuyafujisaki@users.noreply.github.com [flutter_lints] Replace "flutter pub add --dev" with "dev:" (flutter/packages#5260)
2023-10-28 engine-flutter-autoroll@skia.org Roll Flutter from 5907c9757e3f to a4ec6278b1a4 (18 revisions) (flutter/packages#5255)
2023-10-27 hrkadam.92@gmail.com [go_router] Fixes deep-link with no path on cold start (flutter/packages#5113)
2023-10-27 katelovett@google.com [two_dimensional_scrollables] Fix pinned row painting when one axis is reversed (flutter/packages#5187)
2023-10-27 engine-flutter-autoroll@skia.org Roll Flutter from c555599fbb4a to 5907c9757e3f (45 revisions) (flutter/packages#5252)

If this roll has caused a breakage, revert this CL and stop the roller
using the controls here:
https://autoroll.skia.org/r/flutter-packages-flutter-autoroll
Please CC flutter-ecosystem@google.com,rmistry@google.com on the revert to ensure that a human
is aware of the problem.

To file a bug in Flutter: https://github.com/flutter/flutter/issues/new/choose

To report a problem with the AutoRoller itself, please file a bug:
https://issues.skia.org/issues/new?component=1389291&template=1850622

Documentation for the AutoRoller is here:
https://skia.googlesource.com/buildbot/+doc/main/autoroll/README.md

engine-flutter-autoroll committed:
Roll Flutter Engine from 6b9b806151ed to 9de8146d21bf (1 revision) (#137624)

https://github.com/flutter/engine/compare/6b9b806151ed...9de8146d21bf

2023-10-31 skia-flutter-autoroll@skia.org Roll Skia from 817fca468240 to 40223641388b (19 revisions) (flutter/engine#47507)

If this roll has caused a breakage, revert this CL and stop the roller
using the controls here:
https://autoroll.skia.org/r/flutter-engine-flutter-autoroll
Please CC rmistry@google.com,zra@google.com on the revert to ensure that a human
is aware of the problem.

To file a bug in Flutter: https://github.com/flutter/flutter/issues/new/choose

To report a problem with the AutoRoller itself, please file a bug:
https://issues.skia.org/issues/new?component=1389291&template=1850622

Documentation for the AutoRoller is here:
https://skia.googlesource.com/buildbot/+doc/main/autoroll/README.md

engine-flutter-autoroll committed:
Roll Flutter Engine from 9de8146d21bf to 47465eb4fbe6 (1 revision) (#137626)

https://github.com/flutter/engine/compare/9de8146d21bf...47465eb4fbe6

2023-10-31 skia-flutter-autoroll@skia.org Roll Skia from 40223641388b to 065f0d2302ca (2 revisions) (flutter/engine#47509)

If this roll has caused a breakage, revert this CL and stop the roller
using the controls here:
https://autoroll.skia.org/r/flutter-engine-flutter-autoroll
Please CC rmistry@google.com,zra@google.com on the revert to ensure that a human
is aware of the problem.

To file a bug in Flutter: https://github.com/flutter/flutter/issues/new/choose

To report a problem with the AutoRoller itself, please file a bug:
https://issues.skia.org/issues/new?component=1389291&template=1850622

Documentation for the AutoRoller is here:
https://skia.googlesource.com/buildbot/+doc/main/autoroll/README.md

droidbg committed:
Fix. typos (#137465)

## Description

This PR fixes typos in 
- `date_picker.dart`
- `date_picker_theme.dart`
- `dropdown.dart`

engine-flutter-autoroll committed:
Roll Flutter Engine from 47465eb4fbe6 to 4da3e5b12540 (2 revisions) (#137632)

https://github.com/flutter/engine/compare/47465eb4fbe6...4da3e5b12540

2023-10-31 katelovett@google.com Bump goldctl (flutter/engine#47391)
2023-10-31 skia-flutter-autoroll@skia.org Roll Skia from 065f0d2302ca to b3360479118f (4 revisions) (flutter/engine#47510)

If this roll has caused a breakage, revert this CL and stop the roller
using the controls here:
https://autoroll.skia.org/r/flutter-engine-flutter-autoroll
Please CC rmistry@google.com,zra@google.com on the revert to ensure that a human
is aware of the problem.

To file a bug in Flutter: https://github.com/flutter/flutter/issues/new/choose

To report a problem with the AutoRoller itself, please file a bug:
https://issues.skia.org/issues/new?component=1389291&template=1850622

Documentation for the AutoRoller is here:
https://skia.googlesource.com/buildbot/+doc/main/autoroll/README.md

TahaTesser committed:
Update `ColorScheme.light`, `ColorScheme.dark`, `ColorScheme.highContrastLight`, & `ColorScheme.highContrastDark`  constructors docs for Material 3 (#137149)

fixes [Clarify ColorScheme fromSwatch/fromSeed usage](https://github.com/flutter/flutter/issues/132584)

"This explains how to use `ColorScheme.fromSeed` as a substitute for each `ColorScheme` constructor."

| `ColorScheme.light` (left) to `ColorScheme.fromSeed` (right) | 
| --------------- | 
| ![light](https://github.com/flutter/flutter/assets/48603081/e056e723-5640-4b05-8feb-ca6b517c8682)	|

| `ColorScheme.dark` (left) to `ColorScheme.fromSeed` (right) | 
| --------------- | 
| ![dark](https://github.com/flutter/flutter/assets/48603081/5ff32611-bfb6-49ee-a34e-f935f580e84e)	|

| `ColorScheme.highContrastLight` (left) to `ColorScheme.fromSeed` (right) | 
| --------------- | 
| ![highContrastLight](https://github.com/flutter/flutter/assets/48603081/4b47f2e3-ea8e-4148-85cc-69690e9082c7) |

| `ColorScheme.highContrastDark` (left) to `ColorScheme.fromSeed` (right) | 
| --------------- | 
| ![highContrastDark](https://github.com/flutter/flutter/assets/48603081/3dbd7ec4-c78e-4228-a8ed-673832681563) |

engine-flutter-autoroll committed:
Roll Flutter Engine from 4da3e5b12540 to 39be0fc34503 (4 revisions) (#137635)

https://github.com/flutter/engine/compare/4da3e5b12540...39be0fc34503

2023-10-31 matanlurey@users.noreply.github.com [Impeller] OpenGLES MSAA Render Buffers (i.e. for stencils) (flutter/engine#47495)
2023-10-31 flar@google.com [Impeller] Migrate unit tests to named Rect factories (flutter/engine#47430)
2023-10-31 khalidcomilla58@gmail.com [Typo fixed] in DEPS (flutter/engine#47440)
2023-10-31 jonahwilliams@google.com [Impeller] Restore GLES GPU query times. (flutter/engine#47511)

If this roll has caused a breakage, revert this CL and stop the roller
using the controls here:
https://autoroll.skia.org/r/flutter-engine-flutter-autoroll
Please CC rmistry@google.com,zra@google.com on the revert to ensure that a human
is aware of the problem.

To file a bug in Flutter: https://github.com/flutter/flutter/issues/new/choose

To report a problem with the AutoRoller itself, please file a bug:
https://issues.skia.org/issues/new?component=1389291&template=1850622

Documentation for the AutoRoller is here:
https://skia.googlesource.com/buildbot/+doc/main/autoroll/README.md

ksokolovskyi committed:
Merge branch 'master' into text-selection-leak-tracking

ksokolovskyi committed:
Merge branch 'master' into text-selection-leak-tracking

polina-c commented on packages/flutter/lib/src/rendering/editable.dart:
```diff
@@ -894,6 +898,8 @@ class RenderEditable extends RenderBox with RelayoutWhenSystemFontsChangeMixin,
       _showHideCursor();
       _showCursor.addListener(_showHideCursor);
     }
+    _internalShowCursor?.dispose();
```
The only goal of this field is to dispose cursor.
So, the name `_internalShowCursor` does not describe the meaning.
Also, as it is separate from `_showCursor` it is not very clear when the field should be disposed.

So, I still suggest to have a flag like `_disposeShowCursor`, and then it is clear that current value of _showCursor should be disposed if it is not needed any more.

How does it sound?

ksokolovskyi replied:
@polina-c, thanks for the review!
I totally agree with you, `_disposeShowCursor` is self-explanatory.

ksokolovskyi committed:
Fix

ksokolovskyi committed:
Merge branch 'master' into text-selection-leak-tracking

ksokolovskyi commented:
@polina-c, thanks a lot for the review!
